### ISSUE TITLE
Data serialization in SSR mode is not possible

### ISSUE THREAD

**Issue by akaufmann** [Baqend/js-sdk#20]
If you use a SSR framework like [Next.js](https://nextjs.org/) or [Nuxt.js](https://nuxtjs.org/) you can't use the Baqend JS SDK properly. Serialization and therefore correct client side hydration is not possible. This has to do with non-POJOs that can't be serialized on the server.

Because of that the framework triggers another call (client-side) on first render to Baqend to fetch the same data.

tl;dr: I don't know if this will ever be possible because the serialization can't handle functions or non-POJOs.

**Comment by fbuecklers** [comment]
Hi,

thanks for the feedback, we are already discussing how we can improve the SDK here. I'm not familiar with Next.js. Do you know if instances of normal classes are supported? Or must all objects be plain POJOs? 
We have a concept in mind that we remove the proxy like behavior with getters/setters how it is implemented right now and switch to normal instance properties. But I'm not sure if that will fix the issue here. 
Feedback is very welcome for our next SDK iteration.

**Comment by fbuecklers** [comment]
I just want to let you know that we are most done to have pojo like Proxies which also should work with Next.js. If you want to give it a try let me know!

**Comment by fbuecklers** [comment]
This is released with 3.0
